gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala

```diff
@@ -535,8 +535,11 @@ object ExpressionConverter extends SQLConfHelper with Logging {
       ColumnarBroadcastExchangeExec(exchange.mode, newChild)
     }
 
-    if (GlutenConfig.getConf.enableScanOnly) {
-      // Disable ColumnarSubqueryBroadcast for scan-only execution.
+    if (
+      GlutenConfig.getConf.enableScanOnly || !GlutenConfig.getConf.enableColumnarBroadcastExchange
```

Comment:
You disable columnar dpp if the columnar broadcast exchange is disabled. Why does this issue relate to `when broadcastJoin is disabled` ?

Reply:
Because I used the following configuration for the test,  i think the issue should be changed to `when fallback broadcast join and broadcast exchange`
--conf spark.gluten.sql.columnar.broadcastJoin=false
--conf spark.gluten.sql.columnar.broadcastExchange=false

Reply:
I updated the issue description